File: examples/test_base64.py
# Test base64 module mapping  
import base64

def encode_data(data: str) -> str:
    """Encode string to base64"""
    encoded_bytes = base64.b64encode(data.encode('utf-8'))
    return encoded_bytes.decode('utf-8')

def decode_data(encoded: str) -> str:
    """Decode base64 string"""
    decoded_bytes = base64.b64decode(encoded)
    return decoded_bytes.decode('utf-8')

def encode_url_safe(data: str) -> str:
    """Encode string to URL-safe base64"""
    encoded_bytes = base64.urlsafe_b64encode(data.encode('utf-8'))
    return encoded_bytes.decode('utf-8')

File: examples/test_base64.rs
use base64;
#[doc = "Encode string to base64"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn encode_data(data: &str) -> String {
    let encoded_bytes = base64::engine::general_purpose::STANDARD.encode(data.as_bytes().to_vec());
    String::from_utf8_lossy(&encoded_bytes).to_string()
}
#[doc = "Decode base64 string"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn decode_data(encoded: &str) -> String {
    let decoded_bytes = base64::engine::general_purpose::STANDARD
        .decode(encoded)
        .unwrap();
    String::from_utf8_lossy(&decoded_bytes).to_string()
}
#[doc = "Encode string to URL-safe base64"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn encode_url_safe(data: &str) -> String {
    let encoded_bytes = base64::engine::general_purpose::URL_SAFE.encode(data.as_bytes().to_vec());
    String::from_utf8_lossy(&encoded_bytes).to_string()
}
